src/validation_workflow/docker/validation_docker.py

```diff
@@ -0,0 +1,203 @@
+            _OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.dockersource == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
+
+            self.local_image_OS_id = self.get_image_id(_OS_image_name, self.args.version)
+            self.local_image_OSD_id = self.get_image_id(_OSD_image_name, self.args.version)
+            logging.info('the OS image ID is : ' + self.local_image_OS_id)
+            logging.info('the OSD image ID is : ' + self.local_image_OSD_id + '\n\n')
+
+            return True
+
+        else:
+            raise Exception('docker_daemon check is fail. Exiting the validation.')
+
+    def installation(self) -> bool:
+        ''' Pass this method for docker as no installation required in docker'''
+        return True
+
+    def start_cluster(self) -> bool:
+        ''' Pass this method for docker and combine it with the following method because we want to Pass the digest validation in docker first before spinning up the docker container'''
+        return True
+
+    def validation(self) -> bool:
+        # STEP 2 . inspect image digest betwwen opensearchproject(downloaed/local) and opensearchstaging(dockerHub)
+
+        self._OS_inspect_digest = InspectDockerImage(self.local_image_OS_id, self.args.OS_image, self.args.version)
+        self._OSD_inspect_digest = InspectDockerImage(self.local_image_OSD_id, self.args.OSD_image, self.args.version)
+        if self._OS_inspect_digest.inspect_digest() and self._OSD_inspect_digest.inspect_digest():
+            logging.info('Image digest is validated\n\n')
+
+            # STEP 3 . spin-up OS/OSD cluster
+            return_code, self._target_yml_file = self.run_container(
+                self.args.OS_image,
+                self.args.OSD_image,
+                self.args.version,
+                'opensearch-node1-test',
+                'opensearch-node2-test',
+                'opensearch-dashboards-test',
+            )
+            if (return_code):
+                logging.info('Cluster is running now')
+                logging.info('Checking if cluster is ready for API test every 10 seconds\n\n')
+                while True:
+                    if self.check_http_request():
+                        logging.info('\n\ncluster is now ready for API test\n\n')
+                        break
+                    logging.info('sleeping 10sec')
+                    time.sleep(10)
+
+                # STEP 4 . OS, OSD API validation
+                _test_result, _counter = TestCases().test_cases()
+
+                if (_test_result):
+                    logging.info(f'All tests Pass : {_counter}')
+                    return True
+                else:
+                    raise Exception(f'Not all tests Pass : {_counter}')
+            else:
+                raise Exception('The container is fail to run. Exiting the validation.')
```

Comment:
Rewording: `'The container failed to start. Exiting the validation.'`

Reply:
Thanks, @gaiksaya , sounds good, let us update it at next commit.